src/store/app/mutations.ts

```diff
@@ -1,4 +1,11 @@
 import { defineMutations } from 'direct-vuex'
 import { S } from './state'
 
-export const mutations = defineMutations<S>()({})
+export const mutations = defineMutations<S>()({
+  addSearchHistory(state, newHistory: string) {
+    const index = state.searchHistories.indexOf(newHistory)
+    if (index !== -1) state.searchHistories.splice(index, 1)
```

Comment:
```suggestion
    if (index !== -1) {
      state.searchHistories.splice(index, 1)
    }

```
こうしてほしいです :pray: 

- 早期リターン以外の条件分岐は見やすさのためにブロックにしたいです
- この`splice`と`indexOf`は存在してたら消すという意味の二行なので、下とは区切りをつけるために空行を挟みたいです